esp/src/src-react/components/Sasha.tsx

```diff
@@ -1,96 +1,88 @@
+      console.error('Error:', error);
+      setResult(nlsHPCC.noDataMessage);
+    }
+  };
 
   return (
-    <div>
-      <form onSubmit={handleSubmit}>
-        <Dropdown
-          placeholder={nlsHPCC.SelectAnOption}
-          selectedKey={selectedOption}
-          onChange={handleOptionChange}
-          options={[
-            { key: "", text: nlsHPCC.SelectAnOption },
-            { key: "getVersion", text: nlsHPCC.GetVersion },
-            { key: "getLastServerMessage", text: nlsHPCC.GetLastServerMessage },
-            { key: "restoreECLWorkUnit", text: nlsHPCC.RestoreECLWorkunit },
-            { key: "restoreDFUWorkUnit", text: nlsHPCC.RestoreDFUWorkunit },
-            { key: "archiveECLWorkUnit", text: nlsHPCC.ArchiveECLWorkunit },
-            { key: "archiveDFUWorkUnit", text: nlsHPCC.ArchiveDFUWorkunit },
-            { key: "backupECLWorkUnit", text: nlsHPCC.BackupECLWorkunit },
-            { key: "backupDFUWorkUnit", text: nlsHPCC.BackupDFUWorkunit }
-          ]}
-          styles={{ dropdown: { width: 400 } }}
-        />
-        {["restoreECLWorkUnit", "restoreDFUWorkUnit", "archiveECLWorkUnit", "archiveDFUWorkUnit", "backupECLWorkUnit", "backupDFUWorkUnit"].includes(selectedOption) && (
-          <div>
-            <TextField
-              label= {nlsHPCC.WUID}
-              value={wuid}
-              onChange={(event: React.FormEvent<HTMLInputElement>, newValue?: string) => setWuid(newValue || "")}
-              styles={{ fieldGroup: { width: 400 } }}
+    <HolyGrail
+      main={
+        <div>
+          <form onSubmit={handleSubmit}>
+            <Dropdown
+              placeholder={nlsHPCC.SelectAnOption}
+              selectedKey={selectedOption}
+              onChange={handleOptionChange}
+              options={[
+                { key: "", text: nlsHPCC.SelectAnOption },
+                { key: "getVersion", text: nlsHPCC.GetVersion },
+                { key: "getLastServerMessage", text: nlsHPCC.GetLastServerMessage },
+                { key: "restoreECLWorkUnit", text: nlsHPCC.RestoreECLWorkunit },
+                { key: "restoreDFUWorkUnit", text: nlsHPCC.RestoreDFUWorkunit },
+                { key: "archiveECLWorkUnit", text: nlsHPCC.ArchiveECLWorkunit },
+                { key: "archiveDFUWorkUnit", text: nlsHPCC.ArchiveDFUWorkunit },
+                { key: "backupECLWorkUnit", text: nlsHPCC.BackupECLWorkunit },
+                { key: "backupDFUWorkUnit", text: nlsHPCC.BackupDFUWorkunit }
+              ]}
+              styles={{ dropdown: { width: 400 } }}
             />
-          </div>
-        )}
-        <PrimaryButton type="submit" >{nlsHPCC.Submit}</PrimaryButton>
-        {/* Render defaultValue when result is empty */}
-        {defaultValue}
-        {/* Render result when available */}
-        {result && <div>{nlsHPCC.Results}: {result}</div>}
-      </form>
-    </div>
+            {(selectedOption === "archiveECLWorkUnit" || selectedOption === "archiveDFUWorkUnit") && (
```

Comment:
That `[...].includes(selectedOption)` you had earlier makes more sense to me, cause you'll eventually have like 6 === comparisons. Or you could change it to compare against the ones where it shouldn't display the wuid field `(selectedOption !== "getVersion" || selectedOption !== "getLastServerMessage")`. Either way.